certificate-service/domain/src/main/java/se/inera/intyg/certificateservice/domain/validation/model/ElementValidationDateRangeList.java

```diff
@@ -0,0 +1,210 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package se.inera.intyg.certificateservice.domain.validation.model;
+
+import static se.inera.intyg.certificateservice.domain.certificatemodel.model.CheckboxDateRange.RANGE_SUFFIX;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Stream;
+import lombok.AccessLevel;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.Value;
+import se.inera.intyg.certificateservice.domain.certificate.model.DateRange;
+import se.inera.intyg.certificateservice.domain.certificate.model.ElementData;
+import se.inera.intyg.certificateservice.domain.certificate.model.ElementValue;
+import se.inera.intyg.certificateservice.domain.certificate.model.ElementValueDateRangeList;
+import se.inera.intyg.certificateservice.domain.certificatemodel.model.CheckboxDateRange;
+import se.inera.intyg.certificateservice.domain.certificatemodel.model.ElementId;
+import se.inera.intyg.certificateservice.domain.certificatemodel.model.FieldId;
+
+@Value
+@Getter(AccessLevel.NONE)
+@Builder
+public class ElementValidationDateRangeList implements ElementValidation {
+
+  boolean mandatory;
+
+  @Override
+  public List<ValidationError> validate(ElementData data,
+      Optional<ElementId> categoryId) {
+    if (data == null) {
+      throw new IllegalArgumentException("Element data is null");
+    }
+
+    final var dateRangeList = getValue(data.value());
+
+    final var childValidationErrors = getChildValidationErrors(data, categoryId, dateRangeList);
+    if (childValidationErrors != null && !childValidationErrors.isEmpty()) {
+      return childValidationErrors;
+    }
+
+    final var overlappingErrors = getOverlappingErrors(data, categoryId, dateRangeList);
+    if (overlappingErrors != null && !overlappingErrors.isEmpty()) {
+      return overlappingErrors;
+    }
+
+    if (mandatory && !isDateRangeListFilled(dateRangeList)) {
+      return List.of(
+          errorMessage(
+              data,
+              dateRangeList.dateRangeListId(),
+              categoryId,
+              "Välj minst ett alternativ."
+          )
+      );
+    }
+
+    return Collections.emptyList();
+  }
+
+  private List<ValidationError> getChildValidationErrors(ElementData data,
+      Optional<ElementId> categoryId, ElementValueDateRangeList dateRangeList) {
+    if (dateRangeList.dateRangeList() == null) {
+      return Collections.emptyList();
+    }
+
+    final var incompleteErrors = getIncompleteDateRangeErrors(data, categoryId, dateRangeList);
+    final var incorrectErrors = getIncorrectDateRangeErrors(data, categoryId, dateRangeList);
```

Comment:
Maybe we should change the name of this method to better communicate that we are checking if the start date is after the end date?

Reply:
Done